go-controller/pkg/testing/testing.go

```diff
@@ -16,26 +17,63 @@ func init() {
 	format.TruncatedDiff = false
 }
 
+// KCmd is a callback spec returning a k8s exec command
+type KCmd func(cmd string, args ...string) kexec.Cmd
+
 // FakeExec is a convenience struct that wraps testing.FakeExec
 type FakeExec struct {
-	fakeexec.FakeExec
```

Comment:
@alexanderConstantinescu I don't think we actually need to remove this? This struct can still embed fakeexec.FakeExec and access it as it currently does. eg I don't think we need KFake as a separate member.

Reply:
I added `KFake` as I needed a named field for `fakexec.FakeExec`. If I don't give it a name I get a `mixture of field:value and value initializers` when initializing the object in `NewFakeExec`. Maybe there's a way to get around this that I am not thinking about?

Reply:
@alexanderConstantinescu that's odd; leaving it as-is works for me, eg:

```
	return &FakeExec{
		fakeexec.FakeExec{
			LookPathFunc: func(file string) (string, error) {
				return "/fake-bin/" + file, nil
			},
		},
		looseCompare: looseCompare,
		commandPool:  make(map[string]KCmd),
	}
```

Reply:
Really odd. I run go 1.13.4 but I downloaded a 1.12.13 container and ran the tests inside, and I still get the error if I do that. 

```
root@ab50f44eb592:/home# go test ./...
# github.com/ovn-org/ovn-kubernetes/go-controller/pkg/testing
pkg/testing/testing.go:49:20: mixture of field:value and value initializers
?   	github.com/ovn-org/ovn-kubernetes/go-controller/cmd/ovn-k8s-cni-overlay	[no test files]
?   	github.com/ovn-org/ovn-kubernetes/go-controller/cmd/ovn-kube-util	[no test files]
?   	github.com/ovn-org/ovn-kubernetes/go-controller/cmd/ovn-kube-util/app	[no test files]
?   	github.com/ovn-org/ovn-kubernetes/go-controller/cmd/ovnkube	[no test files]
FAIL	github.com/ovn-org/ovn-kubernetes/go-controller/pkg/cluster [build failed]
ok  	github.com/ovn-org/ovn-kubernetes/go-controller/pkg/cni	0.015s
?   	github.com/ovn-org/ovn-kubernetes/go-controller/pkg/cni/types	[no test files]
FAIL	github.com/ovn-org/ovn-kubernetes/go-controller/pkg/config [build failed]
ok  	github.com/ovn-org/ovn-kubernetes/go-controller/pkg/factory	33.506s
?   	github.com/ovn-org/ovn-kubernetes/go-controller/pkg/kube	[no test files]
ok  	github.com/ovn-org/ovn-kubernetes/go-controller/pkg/kube/healthcheck	0.009s
FAIL	github.com/ovn-org/ovn-kubernetes/go-controller/pkg/ovn [build failed]
?   	github.com/ovn-org/ovn-kubernetes/go-controller/pkg/ovn/allocator	[no test files]
FAIL	github.com/ovn-org/ovn-kubernetes/go-controller/pkg/util [build failed]
root@ab50f44eb592:/home# go version
go version go1.12.13 linux/amd64
```
Stackoverflow seems to point to the same issue: https://stackoverflow.com/questions/27626765/mixture-of-fieldvalue-and-value-initializers

Reply:
@alexanderConstantinescu my bad, it should be:

```
// newFakeExec returns a new FakeExec with a default LookPathFunc
func newFakeExec(looseCompare bool) *FakeExec {
	return &FakeExec{
		FakeExec: fakeexec.FakeExec{
			LookPathFunc: func(file string) (string, error) {
				return "/fake-bin/" + file, nil
			},
		},
		looseCompare: looseCompare,
		commandPool:  make(map[string]KCmd),
	}
}
```

So unrelated to embedding, but just the mix of field/value as the error states.

Reply:
Cool! Didn't know you could do that. Just pushed the updated version